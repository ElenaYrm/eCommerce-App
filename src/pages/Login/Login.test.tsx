import { render, screen } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import * as routerHooks from 'react-router-dom';
import { Login } from './index';

jest.mock('react-redux');
jest.mock('react-router-dom');
jest.mock('../../constant/metaData', () => {});
jest.mock('../../services/sdk/auth/methods', () => {});
jest.mock('../../services/sdk/product/methods', () => {});
jest.mock('../../services/sdk/catalog/methods', () => {});
jest.mock('../../services/sdk/customer/methods', () => {});

const useDispatchSpy = jest.spyOn(reduxHooks, 'useDispatch');
const useNavigateSpy = jest.spyOn(routerHooks, 'useNavigate');

describe('Test Login page: ', () => {
  test('should render Login page without error', () => {
    useDispatchSpy.mockReturnValue(jest.fn());
    useNavigateSpy.mockReturnValue(jest.fn());
    const component = render(<Login />);
    expect(screen.getByText('Login')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
});
