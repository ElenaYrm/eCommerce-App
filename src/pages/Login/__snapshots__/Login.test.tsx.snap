// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login page:  should render Login page without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login"
      >
        <div
          class="login__container"
        >
          <div
            class="login__form"
          >
            <div
              class="login__title"
            >
              <span>
                Login
              </span>
              <span>
                to your account
              </span>
            </div>
            <form
              class="form"
              novalidate=""
            >
              <div
                class="field"
              >
                <label
                  class="field__label"
                >
                  <span
                    class="visually-hidden"
                  >
                    email
                  </span>
                  <input
                    class="field__input"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                class="field form__password"
              >
                <label
                  class="field__label"
                >
                  <span
                    class="visually-hidden"
                  >
                    password
                  </span>
                  <input
                    class="field__input"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <button
                    class="button__toggle"
                    type="button"
                  >
                    Show
                  </button>
                </label>
              </div>
              <button
                class="btn"
                type="submit"
              >
                Login ( ^ω^)
              </button>
            </form>
          </div>
        </div>
        <div
          class="login__image"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login"
    >
      <div
        class="login__container"
      >
        <div
          class="login__form"
        >
          <div
            class="login__title"
          >
            <span>
              Login
            </span>
            <span>
              to your account
            </span>
          </div>
          <form
            class="form"
            novalidate=""
          >
            <div
              class="field"
            >
              <label
                class="field__label"
              >
                <span
                  class="visually-hidden"
                >
                  email
                </span>
                <input
                  class="field__input"
                  name="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="field form__password"
            >
              <label
                class="field__label"
              >
                <span
                  class="visually-hidden"
                >
                  password
                </span>
                <input
                  class="field__input"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <button
                  class="button__toggle"
                  type="button"
                >
                  Show
                </button>
              </label>
            </div>
            <button
              class="btn"
              type="submit"
            >
              Login ( ^ω^)
            </button>
          </form>
        </div>
      </div>
      <div
        class="login__image"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
